src/ducks/tracking/browser.jsx

```diff
@@ -0,0 +1,61 @@
+import React, { useContext, createContext } from 'react'
+import memoize from 'lodash/memoize'
+import {
+  getTracker as uiGetTracker,
+  shouldEnableTracking
+} from 'cozy-ui/transpiled/react/helpers/tracker'
+
+const trackerShim = {
+  trackPage: () => {},
+  trackEvent: () => {},
+  push: () => {}
+}
+
+export const getMatomoTracker = memoize(() => {
+  const trackerInstance = uiGetTracker(
+    __PIWIK_TRACKER_URL__,
+    __PIWIK_SITEID__,
+    true, //
+    false
+  )
+
+  if (!trackerInstance) {
+    return trackerShim
+  }
+
+  trackerInstance.push([
+    'setTrackerUrl',
+    'https://matomo.cozycloud.cc/matomo.php'
+  ])
+  trackerInstance.push(['setSiteId', 8])
+
+  return {
+    trackEvent: event => {
+      console.info('Tracking event', event)
+      const { name, action, category, type } = event
+      trackerInstance.push(['trackEvent', category, name, action])
+    },
+    trackPage: pageName => {
+      console.info('Tracking page', pageName)
+      trackerInstance.push(['setCustomUrl', pageName])
+      trackerInstance.push(['trackPageView'])
+    }
+  }
+})
+
+export const getTracker = getMatomoTracker
+
+export const TrackerContext = createContext()
+
+export const TrackerProvider = ({ children }) => {
+  const tracker = getTracker()
```

Comment:
Did you test with an account that disabled tracking? I think that tracker will be empty or undefined

Reply:
We return a shimmed tracker, methods are there but do not do anything. This way, components do not care about that.